### DISCUSSION TITLE
sproto dump 无法正确处理*array()

### DISCUSSION THREAD

**Discussion by mrcaorui** [cloudwu/skynet#1834] [General]
``` 
./core.lua:47: syntax error at [.sproto] line (108)
stack traceback:
        [C]: in function 'error'
        ./core.lua:47: in function <./core.lua:46>
        [C]: in function 'lpeg.match'
        ./core.lua:291: in upvalue 'parser'
        ./core.lua:317: in function 'core.gen_trunk'
        sprotodump.lua:57: in main chunk
        [C]: in ?
```
怀疑是core.lua这一句不能正确匹配的原因？
`local mainkey = "(" * blank0 * name * blank0 * ")"`
我改成
`local mainkey = "(" * blank0 * (name^0) * blank0 * ")"`
看起来就解决了

**Comment by cloudwu** [comment]
请给出出错的 .proto 文件。

**Reply by mrcaorui** [reply]
```
.IdReceiveInfo  {
	id  0 : integer
	received    1 : boolean
}

c2gs_id 1 {
	request {
	}
	response {
        # 可能为空
        info    0 : *IdReceiveInfo()
	}
}
```
脱敏之后的内容

**Comment by cloudwu** [comment]
如果你指数组，你必须写 *array 不能写 *array()

**Reply by mrcaorui** [reply]
> For empty main index likes *array(), the array would be encoded as an unordered map with the first field as key and the second field as value.

是想作为map使用

**Comment by mrcaorui** [comment]
云大回复太快了

**Comment by cloudwu** [comment]
core.lua 指的是哪个文件？

**Reply by mrcaorui** [reply]
3rd/sprotodump/core.lua

**Reply by t0350** [reply]
https://github.com/cloudwu/skynet/blob/master/lualib/sprotoparser.lua#L74 skynet 的是这个代码。别的库里的吧

**Reply by t0350** [reply]
https://github.com/lvzixun/sprotodump/blob/master/core.lua#L67 这边也更新了的。更新下呢

**Reply by mrcaorui** [reply]
十分抱歉耽误二位大佬时间了
